addons/l10n_it/models/l10n_it_account_invoice.py

```diff
@@ -0,0 +1,421 @@
+# -*- coding:utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import base64
+import zipfile
+import io
+import logging
+
+from odoo import api, fields, models, _
+from odoo.tools import float_repr
+from odoo.exceptions import UserError, ValidationError
+from odoo.addons.base.models.ir_mail_server import MailDeliveryException
+
+from datetime import date, datetime
+
+_logger = logging.getLogger(__name__)
+
+DEFAULT_FACTUR_ITALIAN_DATE_FORMAT = '%Y-%m-%d'
+
+PAYMENT_METHODS = [
+    ("MP01", "[MP01] contanti"),
+    ("MP02", "[MP02] assegno"),
+    ("MP03", "[MP03] assegno circolare"),
+    ("MP04", "[MP04] contanti presso Tesoreria"),
+    ("MP05", "[MP05] bonifico"),
+    ("MP06", "[MP06] vaglia cambiario"),
+    ("MP07", "[MP07] bollettino bancario"),
+    ("MP08", "[MP08] carta di pagamento"),
+    ("MP09", "[MP09] RID"),
+    ("MP10", "[MP10] RID utenze"),
+    ("MP11", "[MP11] RID veloce"),
+    ("MP12", "[MP12] RIBA"),
+    ("MP13", "[MP13] MAV"),
+    ("MP14", "[MP14] quietanza erario"),
+    ("MP15", "[MP15] giroconto su conti di contabilità speciale"),
+    ("MP16", "[MP16] domiciliazione bancaria"),
+    ("MP17", "[MP17] domiciliazione postale"),
+    ("MP18", "[MP18] bollettino di c/c postale"),
+    ("MP19", "[MP19] SEPA Direct Debit"),
+    ("MP20", "[MP20] SEPA Direct Debit CORE"),
+    ("MP21", "[MP21] SEPA Direct Debit B2B"),
+    ("MP22", "[MP22] Trattenuta su somme già riscosse")
+]
+
+class AccountInvoice(models.Model):
+    _inherit = 'account.invoice'
+    _name = 'account.invoice'
+
+    send_state = fields.Selection([
+        ('new', 'New'),
+        ('other', 'Other'),
+        ('to_send', 'Not yet send'),
+        ('sent', 'Sent, waiting for response'),
+        ('invalid', 'Sent, but invalid'),
+        ('delivered', 'This invoice is delivered'),
+        ('delivered_accepted', 'This invoice is delivered and accepted by destinatory'),
+        ('delivered_refused', 'This invoice is delivered and refused by destinatory'),
+        ('delivered_expired', 'This invoice is delivered and expired (expiry of the maximum term for communication of acceptance/refusal)'),
+        ('failed_delivery', 'Delivery impossible, ES certify that it has received the invoice and that the file \
+                        could not be delivered to the addressee') # ok we must do nothing
+    ], default='to_send')
+
+    stamp_duty = fields.Float(default=0, string="Dati Bollo", size=15, readonly=True, states={'draft': [('readonly', False)]})
+
+    transport_document_number = fields.Char(string="Numero DDT", size=20, readonly=True, states={'draft': [('readonly', False)]}, help="Transport document number")
```

Comment:
No way to just do a many2many between ddts and invoices and compute the dates from there?

Reply:
Actually, a ddt can just be linked with a many2one to a picking, no need for many2many and everything could be calculated.

Reply:
So, ddt here could just be a manual field, which we inherit when stock is installed.